### DISCUSSION TITLE
Adding Coins

### DISCUSSION THREAD

**Discussion by minershaven** [coinfoundry/miningcore#1310] [Q&A] [answered]
If I want to add a different coin to miningcore and the algorithm is listed in https://github.com/oliverw/miningcore/tree/master/src/Miningcore/Crypto/Hashing

Is it just a matter of adding that coins details to the coins.json file ?

**Comment by oliverw** [comment, marked as answer]
> If I want to add a different coin to miningcore and the algorithm is listed in https://github.com/oliverw/miningcore/tree/master/src/Miningcore/Crypto/Hashing
> 
> Is it just a matter of adding that coins details to the coins.json file ?

If both the hash algorithm and - much more important - the coin family is already supported, then there's a good chance for modifying `coins.json` is all it takes. Some coins come with internal changes that might not be obvious at first glance (not even to me), which require code changes in Miningcore as well.

**Reply by minershaven** [reply]
Thank you for the reply. Looks like both the algorithm and the coin family is there so I will give it a try. Thanks again for the help Oliver!

**Comment by codegatr** [comment]
@oliverw , is there any article about how to add a coin?
Ur Video.

I'm new on it :D

**Reply by oliverw** [reply]
> @oliverw , is there any article about how to add a coin? Ur Video.
> 
> I'm new on it :D

No, there's just hard work and potentially a lot of pain. 😁

**Reply by codegatr** [reply]
Haha, that answer was very nice.
But it's very difficult to fine clean site, to know how to add a coins.
It must be a Panel to be easy.
